### PR TITLE
Community/apiv10

### LINKED ISSUE #467: Tracking for API v10 migration
This reflects the changes that need to / have already been done for v10. Current branch for this is https://github.com/nextcord/nextcord/tree/community/apiv10

For more info about changes see https://github.com/discord/discord-api-docs/discussions/4510

- [x] Bump version in urls
- [x] Message Intent (Will be done in #457)
- [x] `application.summary` is depreceated and should be removed (Will be done in #464 )
- [x] document removal of `application.summary`
- [x] ~~Achievement localization format has changed. name and description are now strings, and localized strings are now stored in name_localizations and description_localizations~~ Game SDK only
- [x] ~~`/channels/<channel_id>/threads/active` has been removed~~ we do not use that
- [x] Existing attachments must be specified when PATCHing messages with new attachments. Any attachments not specified will be removed and replaced with the specified list
- [x] Audit log reason as a body/query parameter is no longer supported, and you should instead use the X-Audit-Log-Reason header.  (Will be done in #466)
- [x] Message routes accept embeds rather than embed (Will be done in #464 )
- [x] Requests to v10 and higher will no longer be supported on discordapp.com 
- [ ] Test the code
- [x] Document in changelog
- [x] Wait until it is actually the recommended API Version

### PR DESCRIPTION
**Pull request by ooliver1**
## Summary

Switches nextcord to use discord api v10 and fixes the [relevant changes](https://discord.com/developers/docs/change-log#api-v10)
in order of changelog
- not used
- not used
- added intent
- changed for kick
- resolved
- was removed in #560
- unused
- unused

closes #467

<u>to test</u>

- [x] message content without intents
- [x] message content with intents
- [x] intents.message_content
- [x] ApplicationFlags.gateway_message_content
- [x] kick
- [x] backport to v9

## Checklist

- [x] If code changes were made then they have been tested.
  - [x] I have updated the documentation to reflect the changes.
  - [x] I have run `task pyright` and fixed the relevant issues.
- [x] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [x] This PR is a breaking change (e.g. methods or parameters removed/renamed) technically? message.content will not resolve
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Comment by ooliver1** [comment]
needs rough testing
- message content intent
- message embeds
- kick with reason

**Comment by Soheab** [comment]
Assuming this is going to get merged before the intent deadline, i think there should be an easy way for users to switch back to v9 (like users that can't enable the intent on the portal for unknown reasons). This can be achieved with a global var and a variable users can set I think, something like `discord.http.API_VERSION = 9`.

**Comment by TAG-Epic** [comment]
> Assuming this is going to get merged before the intent deadline, i think there should be an easy way for users to switch back to v9 (like users that can't enable the intent on the portal for unknown reasons). This can be achieved with a global var and a variable users can set I think, something like `discord.http.API_VERSION = 9`.
> 
> Other than that, looks good to me.

Message content intent still applies on v9.

**Comment by Soheab** [comment]
> Message content intent still applies on v9.

That is true but it's not enforced there (yet?) aka users don't need to enable it on the portal and in code.

**Comment by ooliver1** [comment]
> > Assuming this is going to get merged before the intent deadline, i think there should be an easy way for users to switch back to v9 (like users that can't enable the intent on the portal for unknown reasons). This can be achieved with a global var and a variable users can set I think, something like `discord.http.API_VERSION = 9`.
> > Other than that, looks good to me.
> 
> Message content intent still applies on v9.

Well yeah, but not until August, it applies on v10 fully

**Comment by ooliver1** [comment]
> Assuming this is going to get merged before the intent deadline, i think there should be an easy way for users to switch back to v9 (like users that can't enable the intent on the portal for unknown reasons). This can be achieved with a global var and a variable users can set I think, something like `discord.http.API_VERSION = 9`.
> 
> Other than that, looks good to me.

@Soheab how does https://github.com/nextcord/nextcord/pull/647/files#diff-517d49605d29d0371ebe85c47db30b46a9d5a2b12623eac744ab66d4655f2ed3R116-R128 look

File: docs/migrating_2.rst [hunk 1]
```diff
@@ -264,6 +265,26 @@ Guild.get_member taking positional only argument
...
+You can do this by using :meth:`Intents.all` or by setting :attr:`~Intents.message_content` to ``True``:
+
+.. code-block:: python3
+
+    intents = nextcord.Intents.default()
+    intents.message_content = True
+
+For more information go to the :ref:`message content intent documentation <need_message_content_intent>`.
+
+If you are unable to get the message content intent and would like some time to migrate to application commands:
```
Comment (Skelmis) [inline docs/migrating_2.rst:281]: If you are unable to get the message content intent and would like some time to migrate to application commands you can downgrade the API back to version 9 as shown below.

**Review by Skelmis — CHANGES_REQUESTED** [review]

**Review by Skelmis — APPROVED** [review]

**Pull request merged**